src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -325,10 +327,53 @@ internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink
                     errorSink: errorSink);
         }
 
-        private static bool ValidateName(
-            string name,
-            bool targetingAttributes,
-            ErrorSink errorSink)
+        private static bool ValidateRequiredAttributeName(string attributeName, ErrorSink errorSink)
+        {
+            if (attributeName.Length >= 1 && attributeName[0] == '[')
+            {
+                // Value starts with '[', treat it as a CSS selector.
+
+                if (attributeName[attributeName.Length - 1] != ']')
+                {
+                    errorSink.OnError(SourceLocation.Zero, "TODO: Create error for invalid CSS Selector syntax", length: 0);
+                    return false;
+                }
+
+                var equalsIndex = attributeName.IndexOf('=');
+                if (equalsIndex != -1)
+                {
+                    if (attributeName.IndexOfAny(UnsupportedAttributeSelectors) == equalsIndex - 1)
```

Comment:
Check `attributeName[equalsIndex - 1` against a _whitelist_ of supported selectors.